tutorials/preprocessing/plot_40_artifact_correction_ica.py

```diff
@@ -1,307 +1,549 @@
-#              might not work for ICA processing. Therefore, it is recommended
-#              to use IIR filters for MNE up to 0.14. In MNE 0.15, FIR filters
-#              can be designed with the ``'firwin'`` method, which generally
-#              produces much steeper filters. This method will be the default
-#              FIR design method in MNE 0.16. In MNE 0.15, you need to
-#              explicitly set ``fir_design='firwin'`` to use this method. This
-#              is the recommended filter method for ICA preprocessing.
-
-
-###############################################################################
-# Fit ICA
-# -------
+# .. note::
+#     Before applying ICA (or any artifact repair strategy), be sure to observe
+#     the artifacts in your data to make sure you choose the right repair tool.
+#     Sometimes the right tool is no tool at all — if the artifacts are small
+#     enough you may not even need to repair them to get good analysis results.
+#     See :ref:`tut-artifact-overview` for guidance on detecting and
+#     visualizing various types of artifact.
+#
+#
+# What is ICA?
+# ^^^^^^^^^^^^
+#
+# Independent components analysis (ICA) is a technique for estimating
+# independent source signals from a set of recordings in which the source
+# signals were mixed together in unknown ratios. A common example of this is
+# the problem of `blind source separation`_: with 3 musical instruments playing
+# in the same room, and 3 microphones recording the performance (each picking
+# up all 3 instruments, but at varying levels), can you somehow "unmix" the
+# signals recorded by the 3 microphones so that you end up with a separate
+# "recording" isolating the sound of each instrument?
+#
+# It is not hard to see how this analogy applies to EEG/MEG analysis: there are
+# many "microphones" (sensor channels) simultaneously recording many
+# "instruments" (blinks, heartbeats, activity in different areas of the brain,
+# muscular activity from jaw clenching or swallowing, etc). As long as these
+# various source signals are `statistically independent`_ and non-gaussian, it
+# is usually possible to separate the sources using ICA, and then re-construct
+# the sensor signals after excluding the sources that are unwanted.
+#
+#
+# ICA in MNE-Python
+# ~~~~~~~~~~~~~~~~~
+#
+# MNE-Python implements three different ICA algorithms: ``fastica`` (the
+# default), ``picard``, and ``infomax``. FastICA and Infomax are both in fairly
+# widespread use; Picard is a newer (2017) algorithm that is expected to
+# converge faster than FastICA and Infomax, and is more robust than other
+# algorithms in cases where the sources are not completely independent, which
+# typically happens with real EEG/MEG data. See [1]_ for more information.
+#
+# The ICA interface in MNE-Python mirrors the standard interface in
+# `scikit-learn`_: some general parameters are specified when creating an
```

Comment:
this is not really true, not sure if it's worth mentioning because it may confuse the reader. In `scikit-learn` you have `transform` method but in MNE we use `apply`, so it's not exactly same